### PR TITLE
Modify back_populate_user_levels_level_source_id to process a CSV.

### PR DESCRIPTION
**Pull request by ashercodeorg**
The previous methodology was too slow in practice to perform the backfill.

### CODE REVIEW

**Comment by joshlory** [comment]
Commit message is `Remove rubocop blacklisting of dashboard/db/ directories.`? 😄

EDIT (ASHER): Oops. Fixed. Result of messing up a rebase.

File: bin/oneoff/back_populate_user_levels_level_source_id [hunk 1]
```diff
@@ -1,55 +1,78 @@
...
+    level_source_id = line[2]
+    update_array << [user_id, level_id, level_source_id]
+  end
 
+  # Process the array in batches, so that the transactions are appropriate
+  # sized.
+  update_array.each_slice(batch_size) do |update_array_slice|
     ActiveRecord::Base.transaction do
-      user_level_id_to_level_source_id.each do |user_level_id, level_source_id|
+      update_array_slice.each do |user_id_level_id_level_source_id|
```
Comment (Hamms) [inline bin/oneoff/back_populate_user_levels_level_source_id]: I believe you can just do

```
update_array_slice.each do |user_id, level_id, level_source_id|
  UserLevel.
     where(user_id: user_id)
```

**Comment by Hamms** [comment]
Seems to me that the original version is likely slow because we query the Activity table inidividually inside the `find_each` loop over UserLevel; it might be easier to, rather than creating a new process by which a CSV must be generated, instead just update that query to do the join at the high level so we don't need to make `batch_size` single queries.

That being said, if the work's already done it might not be worthwhile to go back and do it again. If that's the case, this LGTM

**Comment by ashercodeorg** [comment]
The current approach has the advantage that the processing is done on the replica and the saving is done on the master, rather than both on the master. This is a gain, even if there is no increase in speed. The current approach also has the advantage of doing the processing in MySQL, which tends to be faster than ActiveRecord. Given that the work is done, I'm sticking with it.

File: bin/oneoff/back_populate_user_levels_level_source_id (hunk 1 above)
Comment (ashercodeorg) [inline bin/oneoff/back_populate_user_levels_level_source_id]: Done (in next commit).

**Pull request merged**